epflldap/ldap_search.py

```diff
@@ -34,13 +34,12 @@ def is_unit_exist(unit_id):
     Return True if the unit 'unid_id' exists.
     Otherwise return False
     """
-    attribute = 'objectClass'
-    response = LDAP_search(
-        pattern_search="(uniqueidentifier=" + unit_id + ")",
-        attribute=attribute
-    )
-
     try:
+        attribute = 'objectClass'
+        response = LDAP_search(
+            pattern_search="(uniqueidentifier={})".format(unit_id),
+            attribute=attribute
+        )
         return 'EPFLorganizationalUnit' in response[0]['attributes'][attribute]
```

Comment:
soucis avec cette ligne:
- si response est vide -> IndexError
- si le premier élément n'a pas d'attribut 'attributes' -> KeyError
- si le second élément n'a pas d'attribut 'objectClass' -> KeyError

je dis ça car on a déjà eu des IndexError sur le module :)

Reply:
Comme c'est une méthode is_qq_chose, l'idée est  si il y a une erreur = l'unit n'existe pas donc on retourne False.
Les exceptions IndexError et KeyError sont des enfants de Exception donc les exceptions seront catchées.
Je ne comprends pas les soucis que tu décris plus haut ?

Reply:
> je dis ça car on a déjà eu des IndexError sur le module :)

On a eu ces erreurs par le passé, et c'est vrai que le catch exception va tout rattraper. C'est ce que j'ai commenté juste en dessous avec mon 🏇 . Je n'aurai pas du séparer mes deux commentaires -> confusionnant

Reply:
Je suis watcheur de ce repo car je pense l'utiliser à terme, du coup j'ajoute mes two cents sur la question, vous en faites ce que vous voulez ;) 
il me semble que c'est un risque trop important que d'utiliser un try catch Exception ici (et ailleurs aussi).
En effet, si la connexion avec le LDAP tombe, il y aura peut-être une exception générée, et la méthode fonctionnera en répondant "false", alors que ce n'est pas le cas, puisqu'on ne sait pas faute de connexion au LDAP. 
D'ailleurs d'expérience il ne faut que très très rarement utiliser un try catch Exception, car on risque de rendre tout le système d'exception caduque.

J'irais même un peu plus loin en forçant ldap3 de générer un exception pour les cas d'un probème LDAP ( voir http://ldap3.readthedocs.io/installation.html#library-errors)